syntax = "proto3";

option java_multiple_files = true;
option java_package = "distribuidos.proto";
option java_outer_classname = "ServiceProto";

package distribuidos;

service ConvertidorUrls {
  rpc ConvertirUrls (ConvertirUrlsRequest) returns (ConvertirUrlsResponse) {}
}

service ConvertidorOffice {
  rpc ConvertirArchivos (ConvertirArchivosRequest) returns (ConvertirArchivosResponse) {}
}

// Mensaje para representar un elemento URL
message UrlItem {
  string nombre = 1;
  string url = 2;
  string tipo = 3;
}

// Mensaje para representar un elemento Archivo
message ArchivoItem {
  string nombre = 1;
  string contenido_base64 = 2;
  string tipo = 3;
}

message ConvertirUrlsRequest {
  repeated UrlItem urls = 1;
}

message ConvertirUrlsResponse {
  repeated string resultados = 1;
}

message ConvertirArchivosRequest {
  repeated ArchivoItem archivos = 1;
}

message ConvertirArchivosResponse {
  repeated string resultados = 1;
}